### PR TITLE
[AUTO-4758] Update Automation benchmark test to support v2.1

### PR DESCRIPTION
**Pull request by anirudhwarrier**
https://smartcontract-it.atlassian.net/browse/AUTO-4758

Part 1 - Support for Conditional upkeeps. Add StaleReportCount metric.

### CODE REVIEW

File: integration-tests/testreporters/keeper_benchmark.go [hunk 1]
```diff
@@ -101,11 +104,15 @@ func (k *KeeperBenchmarkTestReporter) WriteReport(folderLocation string) error {
...
 	totalReverted = k.NumRevertedUpkeeps
+	totalStaleReports = k.NumStaleUpkeepReports
 	pctWithinSLA := (1.0 - float64(totalMissedSLA)/float64(totalEligibleCount)) * 100
-	var pctReverted float64
+	var pctReverted, pctStale float64
 	if totalPerformed > 0 {
 		pctReverted = (float64(totalReverted) / float64(totalPerformed)) * 100
 	}
+	if totalPerformed > 0 {
+		pctStale = (float64(totalStaleReports) / float64(totalPerformed)) * 100
```
Comment (shileiwill) [inline integration-tests/testreporters/keeper_benchmark.go:114]: Do this math in the if block (line #110) to save some code?

File: integration-tests/testsetups/keeper_benchmark.go [hunk 2]
```diff
@@ -427,6 +441,10 @@ func (k *KeeperBenchmarkTest) ensureInputValues(t *testing.T) {
 	require.NotNil(t, inputs.Upkeeps.FirstEligibleBuffer, "You need to set FirstEligibleBuffer")
 	require.NotNil(t, inputs.RegistryVersions[0], "You need to set RegistryVersion")
 	require.NotNil(t, inputs.BlockTime, "You need to set BlockTime")
+
+	if k.Inputs.DeltaStage == 0 {
+		k.Inputs.DeltaStage = k.Inputs.BlockTime * 5
+	}
```
Comment (shileiwill) [inline integration-tests/testsetups/keeper_benchmark.go:447]: 👍

File: integration-tests/testsetups/keeper_benchmark.go [hunk 3]
```diff
@@ -460,37 +478,47 @@ func (k *KeeperBenchmarkTest) DeployBenchmarkKeeperContracts(
...
-		},
+	var (
+		registry  contracts.KeeperRegistry
+		registrar contracts.KeeperRegistrar
 	)
 
-	// Fund the registry with 1 LINK * amount of AutomationConsumerBenchmark contracts
-	err := k.linkToken.Transfer(registry.Address(), big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(int64(k.Inputs.Upkeeps.NumberOfUpkeeps))))
-	require.NoError(t, err, "Funding keeper registry contract shouldn't fail")
+	if registryVersion <= 3 {
```
Comment (shileiwill) [inline integration-tests/testsetups/keeper_benchmark.go:488]: Nit: I would recommend adding a comment to state which versions are <= 3 or even use the specific versions in code to make code more readable.

**Review by shileiwill — APPROVED** [review]

**Review by ASISBusiness — COMMENTED** [review]
Gn

**Pull request merged**